### PR TITLE
Improve mason build process

### PR DESCRIPTION
**Pull request by benharsh**
This commit updates mason's Makefile to only build mason with:
CHPL_COMM=none
CHPL_LAUNCHER=none
CHPL_REGEXP=re2

It also unloads hugepages if the 'module' command is available. This is an attempt to detect that we're on a Cray, in which case we don't want or need hugepages.

Instead of using 'git -C', use ``chdir`` instead in order to be compatible with older versions of git.

$MASON_HOME/.mason will now be created on the first call to 'mason update' instead of at installation time.

### CODE REVIEW

**Comment by benharsh** [comment]
Latest commit passes test/mason on darwin and linux64.

File: tools/mason/MasonUtils.chpl [hunk 1]
```diff
@@ -75,18 +75,18 @@ proc depExists(dependency: string) {
...
-    }
+  const envHome   = getEnv("MASON_HOME");
+  const home      = getEnv('HOME');
+  const masonHome = if envHome != "" then envHome else home;
+
+  const dotMason = masonHome + "/.mason";
+
+  if isDir(dotMason) == false {
+    writeln(dotMason + " not found, creating...");
+    mkdir(dotMason);
```
Comment (ben-albrecht) [inline tools/mason/MasonUtils.chpl]: Would it be more robust to do a `mkdir(dotMason, parents=true)` here?

File: tools/mason/README.rst [hunk 2]
```diff
@@ -20,25 +20,18 @@ In the mason directory found in ``$CHPL_HOME/tools/mason`` run the following:
...
-dependencies. This installs mason in the same place as the chapel compiler (``chpl``)
-so that mason can be used anywhere in the user's file system.
+It builds the mason binary so that the command line interface can be used. This
+installs mason in the same place as the chapel compiler (``chpl``) so that
+mason can be used anywhere in the user's file system.
 
 
+The ``mason update`` command will download the registry to ``$MASON_HOME/registry``
+if it there is not already a registry at that location.
+
```
Comment (ben-albrecht) [inline tools/mason/README.rst]: Would this be a good place to state the minimum `git` version supported? I guess determining that version is a task in itself..

Comment (benharsh) [reply tools/mason/README.rst]: Probably, but I fixed the "git -C" issue so I think we can punt on it for now.

File: tools/mason/buildMason [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+export CHPL_COMM=none
+export CHPL_LAUNCHER=none
+
+if command -v module > /dev/null 2>&1 ; then
+  echo "Unloading hugepages..."
+  module unload $(module list 2>&1 | \grep "craype-hugepage" | sed -e"s:[0-9]*)::g")
+fi
+
+if ! $chplBinDir/chpl mason.chpl ; then
```
Comment (ben-albrecht) [inline tools/mason/buildMason]: Do we check that `chpl` is in the `$PATH` elsewhere in the module build, and if not, would it be valuable to check it and give a separate error message for that case?

Comment (benharsh) [reply tools/mason/buildMason]: Hmm, I'm not sure. Currently we're planning on people running 'make mason' from $CHPL_HOME, so I feel like it should be OK to use CHPL_BIN_DIR in this way.

Comment (ben-albrecht) [reply tools/mason/buildMason]: Ah, right. Catching this kind of error might be useful elsewhere (like a `make check` for `mason` or something), but definitely not necessary here.

**Pull request merged**